runtime/gas.go

```diff
@@ -4,27 +4,91 @@ import (
 	"context"
 
 	"cosmossdk.io/core/gas"
+	storetypes "cosmossdk.io/store/types"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-var _ gas.Service = (*GasService)(nil)
+var _ gas.Service = GasService{}
 
 type GasService struct{}
 
 func (g GasService) GetGasMeter(ctx context.Context) gas.Meter {
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	return sdkCtx.GasMeter()
+	return CoreGasmeter{gm: sdk.UnwrapSDKContext(ctx).GasMeter()}
 }
 
 func (g GasService) GetBlockGasMeter(ctx context.Context) gas.Meter {
-	return sdk.UnwrapSDKContext(ctx).BlockGasMeter()
+	return CoreGasmeter{gm: sdk.UnwrapSDKContext(ctx).BlockGasMeter()}
 }
 
 func (g GasService) WithGasMeter(ctx context.Context, meter gas.Meter) context.Context {
-	return sdk.UnwrapSDKContext(ctx).WithGasMeter(meter)
+	return sdk.UnwrapSDKContext(ctx).WithGasMeter(SDKGasMeter{gm: meter})
 }
 
 func (g GasService) WithBlockGasMeter(ctx context.Context, meter gas.Meter) context.Context {
-	return sdk.UnwrapSDKContext(ctx).WithBlockGasMeter(meter)
+	return sdk.UnwrapSDKContext(ctx).WithGasMeter(SDKGasMeter{gm: meter})
+}
+
+// ______________________________________________________________________________________________
+// Gas Meter Wrappers
+// ______________________________________________________________________________________________
+
+type SDKGasMeter struct {
```

Comment:
nit: Why not just call this `GasMeter`?

Reply:
the idea was to easily recoginize the difference, since there is a new api for gas metering

Reply:
Is there any other gas meter? We are already in the SDK context here.